src/testFixtures/java/org/opensearch/neuralsearch/BaseNeuralSearchIT.java

```diff
@@ -2884,4 +2884,49 @@ protected void cleanupRemoteModelResources(String connectorId, String modelId) {
             RemoteModelTestUtils.deleteConnector(client(), connectorId);
         }
     }
+
+    /**
+     * Register an agent with the ML Commons plugin
+     *
+     * @param requestBody JSON request body for agent registration
+     * @return agent ID of the registered agent
+     * @throws Exception if registration fails
+     */
+    @SneakyThrows
+    protected String registerAgent(final String requestBody) {
+        Response response = makeRequest(
+            client(),
+            "POST",
+            "/_plugins/_ml/agents/_register",
+            null,
+            toHttpEntity(requestBody),
+            ImmutableList.of(new BasicHeader(HttpHeaders.USER_AGENT, DEFAULT_USER_AGENT))
+        );
+        Map<String, Object> responseJson = XContentHelper.convertToMap(
```

Comment:
we should check is status is 200. Same for agent deletion, that call must be successful/return status 200